use `map_location` with `torch.load` to avoid RuntimeError

CI is failing with the following error anytime any of the `core` functions call `model.load`:
```
    def validate_cuda_device(location):
        device = torch.cuda._utils._get_device_index(location, True)
    
        if not torch.cuda.is_available():
>           raise RuntimeError('Attempting to deserialize object on a CUDA '
                               'device but torch.cuda.is_available() is False. '
                               'If you are running on a CPU-only machine, '
                               'please use torch.load with map_location=torch.device(\'cpu\') '
                               'to map your storages to the CPU.')
E           RuntimeError: Attempting to deserialize object on a CUDA device but torch.cuda.is_available() is False. If you are running on a CPU-only machine, please use torch.load with map_location=torch.device('cpu') to map your storages to the CPU.
```
this is because `model.load` calls `torch.load` but as the `torch` docs say:

> torch.load() uses Python’s unpickling facilities but treats storages, which underlie tensors, specially. They are first deserialized on the CPU and are then moved to the device they were saved from. If this fails (e.g. because the run time system doesn’t have certain devices), an exception is raised. However, storages can be dynamically remapped to an alternative set of devices using the map_location argument.

solution is to 
- [ ] add a device parameter to `model.load` that defaults to `None` which will in turn cause `get_default_device` to be called
- [ ] the `core` functions will pass in whatever device they got, e.g. via `cli` calling them with a config

--- a/src/vak/core/eval.py
+++ b/src/vak/core/eval.py
@@ -138,7 +138,7 @@ def eval(csv_path,
             f'running evaluation for model: {model_name}',
             logger=logger, level='info'
         )
-        model.load(checkpoint_path)
+        model.load(checkpoint_path, device=device)
         metric_vals = model.evaluate(eval_data=val_data,
                                      device=device)
         # create a "DataFrame" with just one row which we will save as a csv;

--- a/src/vak/core/predict.py
+++ b/src/vak/core/predict.py
@@ -189,7 +189,7 @@ def predict(csv_path,
         # ---------------- do the actual predicting --------------------------------------------------------------------
         log_or_print(f'loading checkpoint for {model_name} from path: {checkpoint_path}',
                      logger=logger, level='info')
-        model.load(checkpoint_path)
+        model.load(checkpoint_path, device=device)
         log_or_print(f'running predict method of {model_name}',
                      logger=logger, level='info')
         pred_dict = model.predict(pred_data=pred_data,

--- a/src/vak/engine/model.py
+++ b/src/vak/engine/model.py
@@ -338,7 +338,7 @@ class Model:
             logger=self.logger, level='info')
         torch.save(ckpt, ckpt_path)
 
-    def load(self, ckpt_path):
+    def load(self, ckpt_path, device=None):
         """load model state dict from a checkpoint file.
 
         Loads state_dicts into network and optimizer
@@ -350,10 +350,13 @@ class Model:
         ckpt_path : str, Path
             path including filename from which to load checkpoint
         """
+        if device is None:
+            device = get_default_device()
+
         log_or_print(
             f'Loading checkpoint from:\n{ckpt_path} ',
             logger=self.logger, level='info')
-        ckpt = torch.load(ckpt_path)
+        ckpt = torch.load(ckpt_path, map_location=device)
         self.network.load_state_dict(ckpt['network_state_dict'])
         self.optimizer.load_state_dict(ckpt['optimizer_state_dict'])
 

--- a/tests/test_core/test_learncurve.py
+++ b/tests/test_core/test_learncurve.py
@@ -97,15 +97,24 @@ def test_learncurve(specific_config,
 
 
 def test_learncurve_no_results_path(specific_config,
-                                    tmp_path):
+                                    tmp_path,
+                                    device):
     root_results_dir = tmp_path.joinpath('test_learncurve_no_results_path')
     root_results_dir.mkdir()
 
-    options_to_change = {
-        'section': 'LEARNCURVE',
-        'option': 'root_results_dir',
-        'value': str(root_results_dir)
-    }
+    options_to_change = [
+        {
+            'section': 'LEARNCURVE',
+            'option': 'root_results_dir',
+            'value': str(root_results_dir)
+        },
+        {
+            'section': 'LEARNCURVE',
+            'option': 'device',
+            'value': device
+        }
+    ]
+
     toml_path = specific_config(config_type='learncurve',
                                 audio_format='cbin',
                                 annot_format='notmat',

